src/routes/transactions/mappers/common/transactions.mapper.ts

```diff
@@ -0,0 +1,182 @@
+      if (timestamp !== null) {
+        return this.getDayDate(timestamp, timezoneOffset);
+      } else {
+        throw Error('ExecutionDate cannot be null');
+      }
+    }
+  }
+
+  private groupByDay(
+    transactions: TransactionDomain[],
+    timezoneOffset?: string,
+  ): TransactionDomainGroup[] {
+    return Object.entries(
+      groupBy(transactions, (transaction) => {
+        let transactionTimestamp;
+        if (isMultisigTransaction(transaction)) {
+          transactionTimestamp =
+            transaction.executionDate ?? transaction.submissionDate;
+        } else {
+          transactionTimestamp = transaction.executionDate;
+        }
+        return this.getDayDate(transactionTimestamp, timezoneOffset).getTime();
+      }),
+    ).map(
+      ([timestamp, transactions]) =>
+        <TransactionDomainGroup>{
+          timestamp: Number(timestamp),
+          transactions: transactions,
+        },
+    );
+  }
+
+  private getDayDate(timestamp: Date, timezoneOffset?: string): Date {
+    if (timezoneOffset !== undefined) {
+      timestamp.setUTCSeconds(parseInt(timezoneOffset) || 0);
+    }
+    return new Date(
+      Date.UTC(
+        timestamp.getFullYear(),
+        timestamp.getMonth(),
+        timestamp.getDate(),
+      ),
+    );
+  }
+
+  private mapEthereumTransfer(
+    transfers: Transfer[],
+    chainId: string,
+    safe: Safe,
+  ) {
+    return transfers.map(
+      async (transfer) =>
+        new TransactionItem(
+          await this.incomingTransferMapper.mapTransfer(
+            chainId,
+            transfer,
+            safe,
+          ),
+        ),
+    );
+  }
+
+  private mapGroupTransactions(
+    transactionGroup,
+    chainId,
+    safe,
+  ): Promise<TransactionItem[]> {
+    const transactions: TransactionItem[] = transactionGroup.transactions.map(
+      async (transaction) => {
+        if (isMultisigTransaction(transaction)) {
+          return new TransactionItem(
+            await this.multisigTransactionMapper.mapTransaction(
+              chainId,
+              transaction as MultisigTransaction,
+              safe,
+            ),
+          );
+        } else if (isModuleTransaction(transaction)) {
+          return new TransactionItem(
+            await this.moduleTransactionMapper.mapTransaction(
+              chainId,
+              transaction as ModuleTransaction,
+              safe,
+            ),
+          );
+        } else if (isEthereumTransaction(transaction)) {
+          const transfers = (transaction as EthereumTransaction).transfers;
+          if (transfers != null) {
+            return await Promise.all(
+              this.mapEthereumTransfer(transfers, chainId, safe),
+            );
+          }
+        } else {
+          // This should never happen as AJV would not allow an unknown transaction to get to this stage
+          throw Error('Unrecognized transaction type');
+        }
+      },
+    );
+    return Promise.all(transactions);
```

Comment:
Why do we wrap the `transactions` with a `Promise`? They should all be resolved or? 🤔

Reply:
I changed it to the map function, for some reason is not resolved, take a look 0dac6dc15d88b563a60ab7a8d7d8e6367360f35f